mantidproject/mantid
# Re add rotatesampleshape

### Description of work
This fix was added to not to import and reuse a method from Mantid::DataHandling library by linking with target_link_libraries in cmake which had caused some weird import errors on some Operating systems and to caused to miss some Crystal algorithms.

In this PR, the method that was required to be imported which was a only a small number of lines, was copied into RotateSampleShape class rather than reusing.





#### Further detail of work


### To test:
The problem only appears when using a standalone installer, not a conda install (don't know why). The following build will create standalone installers for all OS, use them to test:
https://builds.mantidproject.org/job/build_packages_from_branch/917/

1- Follow the testing instructions as mentioned in https://github.com/mantidproject/mantid/pull/37908 
2- Check for any startup errors or any missing Crystal libraries in different OSs specially on Linux



Release notes were not added since this is just an internal fix.





---



### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
